templates/ikfast61_moveit_plugin_template.cpp

```diff
@@ -958,6 +977,88 @@ bool IKFastKinematicsPlugin::getPositionIK(const geometry_msgs::Pose &ik_pose,
   return false;
 }
 
+// Used when there are no redundant joints - aka no free params
+bool IKFastKinematicsPlugin::getPositionIK(const std::vector<geometry_msgs::Pose> &ik_poses,
+                                           std::vector< std::vector<double> >& solutions,
+                                           kinematics::KinematicsResult& result,
+                                           const kinematics::KinematicsQueryOptions &options) const
+{
+  ROS_DEBUG_STREAM_NAMED("ikfast","getPositionIK with multiple solutions");
+
+  if(!active_)
+  {
+    ROS_ERROR("kinematics not active");
+    return false;
+  }
+
+  std::vector<double> vfree(free_params_.size());
+  const std::vector<double> ik_seed_state(num_joints_);
+  for(std::size_t i = 0; i < free_params_.size(); ++i)
+  {
+    int p = free_params_[i];
+    ROS_ERROR("%u is %f",p,ik_seed_state[p]);  // DTC
+    vfree[i] = ik_seed_state[p];
+  }
+
+  if(ik_poses.empty())
+  {
+    ROS_ERROR("ik_poses is empty");
+    return false;
+  }
+
+  KDL::Frame frame;
+  tf::poseMsgToKDL(ik_poses[0],frame);
+
+  IkSolutionList<IkReal> ik_solutions;
+  int numsol = solve(frame,vfree,ik_solutions);
+
+  ROS_DEBUG_STREAM_NAMED("ikfast","Found " << numsol << " solutions from IKFast");
+  bool solutions_found = false;
+  if(numsol)
```

Comment:
A quick search didn't turn anything up in the google/ros C++ style guide, but personally I'd prefer `if (numsol > 0)`.

The missing spaces between `if` and `()` are actually mentioned in the style guide.